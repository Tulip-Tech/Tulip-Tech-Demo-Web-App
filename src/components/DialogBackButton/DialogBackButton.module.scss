@use 'src/styles/variables';
@use 'src/styles/theme';

.dialogBackButton {
  position: absolute;
  top: 16px;
  left: 16px;
  width: 48px;
  height: 48px;
}
