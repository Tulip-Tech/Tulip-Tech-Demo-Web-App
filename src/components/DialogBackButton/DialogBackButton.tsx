import React from 'react';

import styles from './DialogBackButton.module.scss';

import IconButton from '#components/IconButton/IconButton';
import ArrowLeft from '#src/icons/ArrowLeft';

type Props = {
  onClick?: () => void;
};

const DialogBackButton: React.FC<Props> = ({ onClick }: Props) => {
  return (
    <IconButton onClick={onClick} className={styles.dialogBackButton} aria-label="Go back">
      <ArrowLeft />
    </IconButton>
  );
};

export default DialogBackButton;
