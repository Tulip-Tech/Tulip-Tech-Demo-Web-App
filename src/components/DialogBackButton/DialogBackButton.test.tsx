import React from 'react';
import { render } from '@testing-library/react';

import DialogBackButton from './DialogBackButton';

describe('<DialogBackButton>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<DialogBackButton />);

    expect(container).toMatchSnapshot();
  });
});
