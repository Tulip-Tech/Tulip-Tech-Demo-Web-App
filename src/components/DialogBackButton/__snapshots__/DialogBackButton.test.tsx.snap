// Vitest Snapshot v1

exports[`<DialogBackButton> > renders and matches snapshot 1`] = `
<div>
  <div
    aria-label="Go back"
    class="_iconButton_0fef65 _dialogBackButton_248199"
    role="button"
    tabindex="0"
  >
    <svg
      aria-hidden="true"
      class="_icon_1e9999"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
        />
      </g>
    </svg>
  </div>
</div>
`;
